src/tests/HDFFile.cxx

```diff
@@ -493,10 +604,308 @@ class T_CommandHandler : public testing::Test {
 
     ASSERT_EQ(recreate_file(&json_command), 0);
   }
+
+  /// Can supply pre-generated test data for a source on a topic to profile the
+  /// writing.
+  class SourceDataGen_f142 {
+  public:
+    string topic;
+    string source;
+    uint64_t seed = 0;
+    std::mt19937 rnd;
+    vector<FlatBufs::f142::fb> fbs;
+    vector<FileWriter::Msg> msgs;
+    // Number of messages already fed into file writer during testing
+    size_t n_fed = 0;
+    /// Generates n test messages which we can later feed from memory into the
+    /// file writer.
+    void pregenerate(size_t array_size, uint64_t n) {
+      LOG(7, "generating {} {}...", topic, source);
+      auto ty = FlatBufs::f142::Value::Double;
+      if (array_size > 0) {
+        ty = FlatBufs::f142::Value::ArrayFloat;
+      }
+      FlatBufs::f142::synth synth(source, ty, int(array_size));
+      rnd.seed(seed);
+      for (uint64_t i1 = 0; i1 < n; ++i1) {
+        // Number of events per message:
+        // size_t n_ele = rnd() >> 24;
+        // Currently fixed, have to adapt verification code first.
+        fbs.push_back(synth.next(i1));
+        auto &fb = fbs.back();
+        msgs.push_back(FileWriter::Msg{(char *)fb.builder->GetBufferPointer(),
+                                       fb.builder->GetSize()});
+      }
+    }
+  };
+
+  static void data_f142() {
```

Comment:
Enormous function with many separate codeblocks and repeated variable names in different scopes...

Reply:
Sure, for future PR.